### ISSUE TITLE
Image#negate generates different image with ImageMagick 7

### ISSUE THREAD

**Issue by Watson1978** [rmagick/rmagick#894]
### Test code
```ruby
require 'rmagick'

image = Magick::ImageList.new('./Flower_Hat.jpg').first
image.negate(true).write('./negate.png')

```

### Result
with IM6 | with IM7
-- | --
<img src="https://user-images.githubusercontent.com/199156/71640597-38717200-2cd0-11ea-8883-1d27ff5957b3.png"> | <img src="https://user-images.githubusercontent.com/199156/71640600-46bf8e00-2cd0-11ea-9ec6-cf45a2fab749.png">

**Comment by Watson1978** [comment]
I think this is bug of ImageMagick 7.

`NegateImage()` API has following description.

```
%    o grayscale: If MagickTrue, only negate grayscale pixels within the image.
```

So, `NegateImage()` API should skip **non grayscale** pixel with `grayscale == MagickTrue`.
However, the following code in `NegateImage()` has skipped with **grayscale** pixel. The conditions should be inverted.
```c
          if (IsPixelGray(image,q) != MagickFalse)
            {
              q+=GetPixelChannels(image);
              continue;
            }
```
https://github.com/ImageMagick/ImageMagick/blob/9e602b5d8ce800a0e36ee45fa111ff32cde18961/MagickCore/enhance.c#L3979-L3983

@dlemstra How do you think about this ?

**Comment by dlemstra** [comment]
Thanks for reporting this. I just pushed a patch to the ImageMagick repository and this issue will be resolved in ImageMagick 7.0.9-14.

**Comment by Watson1978** [comment]
I confirmed this issue was solved with ImageMagick 7 master code.
